### ISSUE TITLE
Review and improve gslab_scons unit tests

### ISSUE THREAD

**Issue by M-R-Sullivan** [gslab-econ/gslab_python#45]
In #36 (specifically, in [this comment](https://github.com/gslab-econ/gslab_python/pull/36#pullrequestreview-15982005)), we noted that the unit tests for `gslab_python`'s `gslab_scons` package should be reviewed and improved. The goal of this issue is to do so. 

The tests currently take over half a minute to run. We should probably find a way to make them run faster.

**Comment by M-R-Sullivan** [comment]
Update: this is under peer review at #49

**Comment by M-R-Sullivan** [comment]
Still under PR at #49.

**Comment by M-R-Sullivan** [comment]
Still under PR at #49.

**Comment by M-R-Sullivan** [comment]
I merged this issue's changes into `master` after finishing peer review at #49.

**Summary**
- I expanded the coverage of `gslab_scons`'s tests.
- I mocked out most external dependencies of `gslab_scons`'s.
- I identified [these items](https://github.com/gslab-econ/gslab_python/pull/49#issuecomment-284099311) as potential problems with `gslab_scons` that we may want to consider addressing at some point. 
- This issue also changed how we run `gslab_python` tests. Instead of running a package's `run_all_tests.py` script, we now use `python setup.py test` at the root of the repository. This command produces a `coverage` report for our testing.
